mongorestore/mongorestore_test.go

```diff
@@ -723,3 +979,48 @@ func TestAutoIndexIdNonLocalDB(t *testing.T) {
 		}
 	})
 }
+
+// TestSkipSystemCollections asserts that certain system collections like "config.systems.sessions" and the transaction
+// related tables aren't applied via applyops when replaying the oplog.
+func TestSkipSystemCollections(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	ctx := context.Background()
+
+	Convey("With a test MongoRestore", t, func() {
+		session.Database("db3").RunCommand(ctx, bson.D{
+			{"create", "c1"},
+		})
+
+		args := []string{
+			DirectoryOption, "testdata/oplog_partial_skips",
+			OplogReplayOption,
+			DropOption,
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+		c1 := session.Database("db3").Collection("c1")
+		c1.Drop(nil)
+
+		// Run mongorestore
+		collBeforeRestore := session.Database("db3").Collection("c1")
+		So(err, ShouldBeNil)
+
+		result := restore.Restore()
+		So(result.Err, ShouldBeNil)
+		So(result.Failures, ShouldEqual, 0)
+
+		// Verify restoration
+		_, err = c1.CountDocuments(nil, bson.M{})
+		So(err, ShouldBeNil)
+
+		assertCollectionEqual(t, collBeforeRestore, session.Database("db3").Collection("c1"), session, ctx)
```

Comment:
I don't think this is correct. collBeforeRestore and the `session.Database("db3").Collection("c1")` are essentially the same thing. They are both the handle for the same collection instead of a snapshot. I think what you should compare is the source collection (the collection from which the dump file was created ) and the destination collection. 
I guess you can bring up a source cluster and then create the dump file in the test. Then use the same file to restore to destin cluster. In this way, you can compare these two collections side by side.

Reply:
Oh right. I'll create two standalones then!